File: Cano-Rodriguez-Claudio/Procesos/Python/pipe.py
import os

def main():
    r, w = os.pipe()  # Create a pipe
    message = "abcde"

    pid = os.fork()
    if pid == 0:  # Child process
        os.close(w)  # Close write end
        rbuf = os.read(r, 5).decode()
        print(f"H: {rbuf}")
        os.close(r)
    else:  # Parent process
        os.close(r)  # Close read end
        os.write(w, message.encode())
        print(f"P: {message}")
        os.close(w)

if __name__ == "__main__":
    main()

File: Cano-Rodriguez-Claudio/Procesos/Go/pipe.go
package main

import (
	"fmt"
	"os"
)

func main() {
	pr, pw, _ := os.Pipe() // Create a pipe
	message := "abcde"
	buffer := make([]byte, 5)

	if pid, _ := os.ForkExec("/proc/self/exe", nil, nil); pid == 0 { // Child process
		pw.Close()              // Close write end
		pr.Read(buffer)         // Read from pipe
		fmt.Printf("H: %s\n", string(buffer))
		pr.Close()
	} else { // Parent process
		pr.Close()              // Close read end
		pw.Write([]byte(message))
		fmt.Printf("P: %s\n", message)
		pw.Close()
	}
}
